### PR TITLE
Fix JavaLambdaHandlerResolver and disable run-markers in tests/inject…

### PR DESCRIPTION
**Pull request by rli**
…ed text

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/src/software/aws/toolkits/jetbrains/utils/PsiUtils.kt [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+package software.aws.toolkits.jetbrains.utils
+
+import com.intellij.injected.editor.VirtualFileWindow
+import com.intellij.openapi.roots.ProjectRootManager
+import com.intellij.psi.PsiElement
+
+fun PsiElement.isTestOrInjectedText(): Boolean {
+    val project = this.project
+    val virtualFile = this.containingFile.virtualFile
+    if (virtualFile is VirtualFileWindow) {
```
Comment (manodnyab) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/utils/PsiUtils.kt:13]: Can the 2 ifs be clubbed with an or?

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/java/JavaLambdaHandlerResolver.kt [hunk 2]
```diff
@@ -39,9 +39,20 @@ class JavaLambdaHandlerResolver : LambdaHandlerResolver {
...
+                                .filter {
+                                    val file = it.containingFile.virtualFile
+
+                                    return@filter if (psiClass.implementsLambdaHandlerInterface(file)) {
+                                        true
+                                    } else {
+                                        it.isValidHandler(psiClass, file)
+                                    }
+                                }
+                        }
```
Comment (manodnyab) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/java/JavaLambdaHandlerResolver.kt:54]: Should the order of the two filters be reversed?

File: jetbrains-core/src/software/aws/toolkits/jetbrains/utils/PsiUtils.kt (hunk 1 above)
Comment (rli) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/utils/PsiUtils.kt:13]: yes, but the second condition is long so optimizing for readability

File: jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/java/JavaLambdaHandlerResolver.kt (hunk 2 above)
Comment (rli) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/java/JavaLambdaHandlerResolver.kt:54]: the second filter is a more expensive check, so cheaper one is first

**Review by abrooksv — APPROVED** [review]

**Pull request merged**